wal/wal.go

```diff
@@ -681,20 +681,19 @@ func NewSegmentsRangeReader(sr ...SegmentRange) (io.ReadCloser, error) {
 
 // segmentBufReader is a buffered reader that reads in multiples of pages.
 // The main purpose is that we are able to track segment and offset for
-// corruption reporting.
+// corruption reporting.  We have to be careful not to increment curr too
+// early.
```

Comment:
What do you mean by `too early.`?

Reply:
> previously it would increment `curr` on the call to `Read` which failed, causing the returns `CorruptionErr` to reference the wrong segment.

I'll add this to the comment.